server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -371,18 +378,15 @@ private boolean registerRepository(RepositoryMetaData repositoryMetaData) throws
                 return false;
             }
         }
-        Repository newRepo = createRepository(repositoryMetaData);
+        createRepository(repositoryMetaData);
         if (previous != null) {
             closeRepository(previous);
```

Comment:
that looks dangerous now (i.e., closing the existing repo). You should rather close the temporarily created repository.

Reply:
Done, the existing one is closed in `org.elasticsearch.repositories.RepositoriesService#applyClusterState` anyway as far as I can see :)